app_dart/test/service/branch_service_test.dart

```diff
@@ -53,6 +54,22 @@ void main() {
       when(githubService.github.repositories).thenReturn(mockRepositoriesService);
     });
 
+    test('return 404 when branch version file does not exist', () async {
+      final gh.Branch candidateBranch = generateBranch(3, name: 'flutter-3.4-candidate.5', sha: '789dev');
+      when(mockRepositoriesService.listBranches(any)).thenAnswer((Invocation invocation) {
+        return Stream.fromIterable([candidateBranch]);
+      });
+      when(mockRepositoriesService.getContents(any, any)).thenAnswer((Invocation invocation) {
+        throw Exception('404 file not found');
```

Comment:
Instead of using generic exceptions, we should throw what is likely to be thrown

```suggestion
        throw GitHubError(github, '404 file not found');
```

Reply:
Oh TIL about `GithubError`